File: laptop-store-backend/model/src/main/java/org/example/model/entity/user/UserDetail.java
package org.example.model.entity.user;

import lombok.Data;
import org.example.model.type.Gender;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "user_detail")
@Data
public class UserDetail implements Serializable {
    @Id
    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "pending_email", length = 100)
    private String pendingEmail;

    @Column(name = "verified_email", length = 100)
    private String verifiedEmail;

    @Column(name = "name", length = 30)
    private String name;

    @Column(name = "phone", length = 10)
    private String phone;

    @Column(name = "gender", length = 10)
    @Enumerated(EnumType.STRING)
    private Gender gender;

    @Column(name = "cart", length = 1000, columnDefinition = "json")
    private String cart;

    @Column(name = "wish_list", length = 1000, columnDefinition = "json")
    private String wishList;

    @Column(name = "address_id")
    private Long addressId;
}

File: laptop-store-backend/model/src/main/kotlin/org/example/model/entity/user/UserDetail.kt
package org.example.model.entity.user

import org.example.model.type.Gender
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "user_detail")
class UserDetail : Serializable {
    companion object {
        private const val serialVersionUID: Long = 1L
    }

    @Id
    @OneToOne
    @JoinColumn(name = "user_id")
    var user: User? = null

    @Column(name = "pending_email", length = 100)
    var pendingEmail: String? = null

    @Column(name = "verified_email", length = 100)
    var verifiedEmail: String? = null

    @Column(name = "name", length = 30)
    var name: String? = null

    @Column(name = "phone", length = 10)
    var phone: String? = null

    @Column(name = "gender", length = 10)
    @Enumerated(EnumType.STRING)
    var gender: Gender? = null

    @Column(name = "cart", length = 1000, columnDefinition = "json")
    var cart: String? = null

    @Column(name = "wish_list", length = 1000, columnDefinition = "json")
    var wishList: String? = null

    @Column(name = "address_id")
    var addressId: Long? = null
}